### PR TITLE
Fix locales handling

### LINKED ISSUE #2572: [Bug] Translation function is missing on the footer
**Describe the bug**
It seems like translated text doesn't correctly output so it only show the function of that text

**Steps to Reproduce**
Scroll down to the bottom

**Logs**
No need.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by SamantazFox**
Fixes #2572
Solves item "Use the locales full name instead of ISO codes" of #1951

Other than that, this changes how the "locale" is passed around:
- Previously, `translate()` would take a `Hash(String, JSON::Any)` as an argument, which implied that every function would set `locale = LOCALES["ISO_code"]`, or in essence, a copy of a complete locale dataset (keys + translation strings). This made the whole Hash being passed around, the entire object is copied to the stack on each function call, and probably impacting performance, among adding unnecessary complexity.
- Now, only the ISO code string is passed, and `translate()` will, by itself, look into the `LOCALES` constant (which is a `Hash(String, Hash(String, JSON::Any))`).

### CODE REVIEW

**Review by syeopite — APPROVED** [review]

**Comment by SamantazFox** [comment]
I've deployed it on https://test.invidious.io and there are quite a few strings being `translate()`'d but which are missing in the `en-US.json` file.

So far:
```patch
diff --git a/locales/en-US.json b/locales/en-US.json
index 036c22f4..d58f7cf7 100644
--- a/locales/en-US.json
+++ b/locales/en-US.json
@@ -433,5 +433,26 @@
     "footer_source_code": "Source code",
     "footer_original_source_code": "Original source code",
     "footer_modfied_source_code": "Modified Source code",
-    "adminprefs_modified_source_code_url_label": "URL to modified source code repository"
+    "adminprefs_modified_source_code_url_label": "URL to modified source code repository",
+    "Embed": "Embed",
+    "Embed Link": "Embed Link",
+    "Subtitles - `x` (.vtt)": "Subtitles - `x` (.vtt)",
+    "dash": "dash",
+    "hd720": "hd720",
+    "medium": "medium",
+    "small": "small",
+    "auto": "auto",
+    "best": "best",
+    "worst": "worst",
+    "4320p": "4320p",
+    "2160p": "2160p",
+    "1440p": "1440p",
+    "1080p": "1080p",
+    "720p": "720p",
+    "480p": "480p",
+    "360p": "360p",
+    "240p": "240p",
+    "144p": "144p",
+    "none": "none",
+    "invidious": "Invidious"
 }
```

I have to see to standardize some of these first, so markking this PR as "blocked" meanwhile.

File: locales/en-US.json [hunk 1]
```diff
@@ -449,5 +450,9 @@
     "footer_source_code": "Source code",
     "footer_original_source_code": "Original source code",
     "footer_modfied_source_code": "Modified Source code",
-    "adminprefs_modified_source_code_url_label": "URL to modified source code repository"
+    "adminprefs_modified_source_code_url_label": "URL to modified source code repository",
+    "Embed": "Embed",
+    "Embed Link": "Embed Link",
+    "Subtitles - `x` (.vtt)": "Subtitles - `x` (.vtt)",
```
Comment (syeopite) [inline locales/en-US.json:456]: While you're at it, could you normalize these keys?

**Comment by SamantazFox** [comment]
Ah, it's also missing:
* `purchased` (search page)
* `360` (ditto)
* `Started streaming 'x' ago` (on a live video)
* `'x' created playlists` (on `/feed/playlists`)
* `'x' saved playlists` (same)

And I think I have done all the pages I'm aware of.

File: locales/en-US.json (hunk 1 above)
Comment (SamantazFox) [reply locales/en-US.json:456]: yup. Will do!

**Comment by SamantazFox** [comment]
Here, done.

NB: I didn't normalize some keys on purpose:
* for the search filters in order to not collide with @syeopite rewrite
* for `Video unavailable` because this type of errors are directly returned from youtube API.

**Comment by SamantazFox** [comment]
final list of missing translations:

```
[warn] i18n: Missing translation key "auto"
[warn] i18n: Missing translation key "Broken? Try another Invidious Instance!"
[warn] i18n: Missing translation key "Cantonese (Hong Kong)"
[warn] i18n: Missing translation key "Chinese"
[warn] i18n: Missing translation key "Chinese (China)"
[warn] i18n: Missing translation key "Chinese (Hong Kong)"
[warn] i18n: Missing translation key "Chinese (Taiwan)"
[warn] i18n: Missing translation key "Could not extract playlistSidebarRenderer."
[warn] i18n: Missing translation key "Dutch (auto-generated)"
[warn] i18n: Missing translation key "English (United Kingdom)"
[warn] i18n: Missing translation key "English (United States)"
[warn] i18n: Missing translation key "Error encountered while retrieving playlist videos.<br>Missing hash key: "playlistVideoListRenderer""
[warn] i18n: Missing translation key "Forced"
[warn] i18n: Missing translation key "French (auto-generated)"
[warn] i18n: Missing translation key "German (auto-generated)"
[warn] i18n: Missing translation key "Indonesian (auto-generated)"
[warn] i18n: Missing translation key "Interlingue"
[warn] i18n: Missing translation key "Italian (auto-generated)"
[warn] i18n: Missing translation key "Japanese (auto-generated)"
[warn] i18n: Missing translation key "Korean (auto-generated)"
[warn] i18n: Missing translation key "Portuguese (auto-generated)"
[warn] i18n: Missing translation key "Portuguese (Brazil)"
[warn] i18n: Missing translation key "Russian (auto-generated)"
[warn] i18n: Missing translation key "Sign in if you've been granted access to this video"
[warn] i18n: Missing translation key "Spanish (auto-generated)"
[warn] i18n: Missing translation key "Spanish (Mexico)"
[warn] i18n: Missing translation key "Spanish (Spain)"
[warn] i18n: Missing translation key "Turkish (auto-generated)"
[warn] i18n: Missing translation key "Vietnamese (auto-generated)"
```

**Review by matthewmcgarvey — APPROVED** [review]

**Comment by SamantazFox** [comment]
@matthewmcgarvey Thanks for your review :D

**Pull request merged**